src/Redux/Crypto.js

```diff
@@ -0,0 +1,27 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const initialState = [];
+const URL = 'https://api.coincap.io/v2/assets';
+
+export const getCryptos = createAsyncThunk('getting Api from cryptos', async () => {
+  const gottenCryptos = await axios.get(URL).catch((error) => error);
+  const cryptoData = gottenCryptos.data;
+  return cryptoData.data;
+});
+
+export const searchCryptos = createAsyncThunk('search the cryptos', async (id) => {
+  const exploreCryptos = await axios.get(`https://api.coincap.io/v2/assets/${id}`).catch((error) => error);
+  return exploreCryptos.data;
+});
```

Comment:
[OPTIONAL]

- For your app to follow the Single Responsibility Principle, I would recommend you make your API call in a separate file and then import it here so that should in case your app grows in the future, it will be easier to maintain 😄 .